controllers/usersController.js

```diff
@@ -0,0 +1,77 @@
+/* eslint-disable class-methods-use-this */
+const { Users } = require('../models');
+
+class UsersController {
+  getAll(req, res) {
+    Users.find()
+      .then((data) => {
+        res.status(200).json(data);
+      })
+      .catch((err) => {
+        res.status(500).json({
+          message: err
+        });
+      });
+  }
+
+  get({ params: { id } }, res) {
+    Users.findOne({
+      _id: id
+    })
+      .then((data) => {
+        res.status(200).json(data);
+      })
+      .catch((err) => {
+        res.status(404).json({
+          message: err
+        });
+      });
+  }
+
+  add({ body: { name, type } }, res) {
+    const user = new Users({
+      name,
+      type
+    });
+    user.save()
+      .then((data) => {
+        res.status(200).json(data);
+      })
+      .catch((err) => {
+        res.status(500).json({
+          message: err
+        });
+      });
+  }
+
+  update({ params: { id }, body: { name } }, res) {
+    Users.updateOne(
+      { _id: id },
+      { $set: { name } }
+    )
+      .then((updateUser) => {
+        res.status(200).json(updateUser);
+      })
+      .catch((err) => {
```

Comment:
Here you should validate that the error is in fact a not found if it is then a 404 is suitable if not then return a 500 error. 

Another solution would be to create functions for notFound and server error that receive the res object and return a user-friendly message and log the real issue to your log.

```js
notFound(res, error, method) {
logger.error(`${error} found in ${method}`);
return res.status(404).json({ message: error });
}
// same for serverError...
```